### ISSUE TITLE
Users In Database Break Relationship Save

### ISSUE THREAD

**Issue by danielml01** [Stillat/relationships#27]
Error:
in_array(): Argument #2 ($haystack) must be of type array, string given
at vendor/stillat/relationships/src/Processors/RelationshipProcessor.php:409

This is because if users are in a database (as opposed to flat files), the relationship field is stored as a string that looks like ["slug", "slug2", "slug3"]

Can be resolved before line 409 with something like:
```
if (is_string($rightReference)) {
    $rightReference = str_ireplace(['[', ']', '\\', '"'], '', $rightReference);
    $rightReference = explode(',', $rightReference);
}
```

**Comment by JohnathonKoster** [comment]
Thanks for the info!

Feel free to open a PR with the proposed change and I can take a look later today - if not, I will likely have time to reproduce/address in the later half of next week.

**Comment by JohnathonKoster** [comment]
Carved out some time for this today and released as part of 2.1.2

**Comment by GioChocolateBro** [comment]
@JohnathonKoster I was using a patch file to make things work with the eloquent driver

```
--- /dev/null
+++ ../src/Processors/RelationshipProcessor.php
@@ -183,7 +183,8 @@
         $updated = [];
 
         if ($this->pristineEntry != null && $this->isNewEntry == false) {
-            $pristine = $this->getFieldValue($fieldName, $this->pristineEntry, []);
+            //$pristine = $this->getFieldValue($fieldName, $this->pristineEntry, []);
+            $pristine = $this->pristineEntry->model()[$fieldName];
         }
 
         if ($this->isDelete) {
```

I don't think I need this anymore right?
